### PR TITLE
Update api.js

### PR DESCRIPTION
**Pull request by guptatrisha97**
Added code to prohibit defining API method twice

### CODE REVIEW

File: src/api.js [hunk 1]
```diff
@@ -782,6 +782,13 @@ API.prototype.declare = function(options, handler) {
     options.output = this._options.schemaPrefix + options.output;
   }
   this._entries.push(options);
+  var valueArr = options.map((item) => {return item.name;});
```
Comment (djmitche) [inline src/api.js]: `options` here is an object with properties -- there's no need to use `map` to find its name attribute.  Just write `options.name`.

File: src/api.js [hunk 2]
```diff
@@ -782,6 +782,13 @@ API.prototype.declare = function(options, handler) {
...
+  var isDuplicate = valueArr.some(function(item, idx) {
```
Comment (djmitche) [inline src/api.js]: This seems to be searching for a duplicate entry in `valueArr`, which is an array of .. well, I'm not sure what map returns when run over an object.  I think what you want here is `this._entries`, which has the list of entries that have already been defined.

File: src/api.js [hunk 3]
```diff
@@ -782,6 +782,13 @@ API.prototype.declare = function(options, handler) {
     options.output = this._options.schemaPrefix + options.output;
   }
   this._entries.push(options);
+  var valueArr = options.map((item) => {return item.name;});
+  var isDuplicate = valueArr.some(function(item, idx) { 
+    return valueArr.indexOf(item) != idx;
```
Comment (djmitche) [inline src/api.js]: You will want to see if there is an item in `this._entries` which has the name `options.name`.  You'll need to do that *before* `options` is pushed onto `this._entries`, though.

File: src/api.js (hunk 2 above)
Comment (guptatrisha97) [reply src/api.js]: in that case, where will I check for options.name? Would it be var isDuplicate = this._entries.some(function(options.name, idx) ?

Comment (djmitche) [reply src/api.js]: You'll check that below, as I suggested.

**Comment by djmitche** [comment]
Picked up by @kritisingh1 in #95

**Pull request closed without merging**